src/west/app/project.py

```diff
@@ -1481,6 +1481,66 @@ def do_add_parser(self, parser_adder):
     def do_run(self, args, user_args):
         self.inf(PurePath(self.topdir).as_posix())
 
+class Change(_ProjectCommand):
+    def __init__(self):
+        super().__init__(
+            'change',
+            'change project details in the manifest',
+            textwrap.dedent('''\
+            Update projects information',
+            'Change project config in west.yml.'''))
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help, 
+            description=self.description
+        )
+
+        parser.add_argument(
+            "name",
+            help="Which project to change.",
+        )
+        parser.add_argument(
+            "--revision",
+            help="What revision to use.",
+        )
+        parser.add_argument(
+            "--path",
+            help="What path to use.",
+        )
+        parser.add_argument(
+            "--url",
+            help="What url to use.",
+        )
+        return parser
+
+        manifest = Manifest.from_file(import_flags=ImportFlag.IGNORE_PROJECTS)
+        m = ruamel.yaml.YAML()
+        m.indent(mapping=2, sequence=4, offset=2)
+        m.preserve_quotes = True
+
+        with open(manifest.path, "r") as fh:
+            x = m.load(fh.read())
+            project_found = False
+            for p in x["manifest"]["projects"]:
+                if p["name"] == args.name:
+                    project_found = True
+                    if args.revision:
+                        p["revision"] = args.revision
+                    if args.path:
+                        print("path")
```

Comment:
please add more meaningful messages like:
"Path of project X changed, Old: ABC New: DEF"